### PR TITLE
LR1121 - Adds FEC to 2.4GHz FSK

### PR DESCRIPTION
**Pull request by JyeSmith**
This PR adds Forward Error Correction (FEC) to the LR1121 when using 2.4GHz FSK. 

FEC is already baked into OTA modulation like LoRa and FLRC, but not something seen in FSK.  For background, FLRC is actually using Minimum Shift Keying (MSK), which is FSK with a particularly set of parameters.  Then FLRC has FEC and Interleaving added by the Semtech transceiver.  This PR essentially is rebuilding FLRC for the LR1121... but without the parameters required for MSK.

The below data is for 2.4GHz K1000, which is the 1000Hz OTA mode used for DK500 and future DK250 mode e.g. these modes are a direct comparison to F1000, D500 and D250.

The 3 FLRC receivers used in the below tests were selected from a group of 6, and the best performing 3 of the batch!

The first 2 plots show packets per second, and have been captured with a single K1000 Tx connected to 3 LR1121 receivers.  Plus a single F1000 Tx connected to 3 SX1280 receivers.  3 Gemini Tx modules where used to generate F1000 interference (simulating 6 separate modules).  To make things worse the FHSS hops have been reduced from 80 to 40 making the domain more congested.  This should be a good representation of a race event.  RSSI was kept high (~ -50dBm) so that only interference and FEC is being assessed.

Baseline capture of the raw FSK 300b/s with no FEC or Interleaving.
image

FSK with FEC and Interleaving.
image

Walking range test to assess FSK LQ against FLRC.  All Tx modules are set to 10mW for testing.
image

To date 2.4GHz FSK has shown a lot of promise, both in close range using DK500 and long range (multiple kilometers) using K1000.  The addition of FEC is looking really promising!  Coupling it with D modes should make it ideal for racing and low latency applications. 

Processing time - It takes 7us to decode a received packet and it can be seen there is a slight delay after processing the first packet of a diversity rx, until the second packets processing begins (the separation in Busy traces). There is no delay introduced processing the second packet.  Encoding before a transmission takes about 10us.  These time could be reduced further with some 8x8b transposing magic for the interleaving, and is something Im looking into.
image

### CODE REVIEW

**Review by pkendall64 — COMMENTED** [review]
Need to add FEC to the `lib_ignore` line in `platformio.ini` so t's not built for unit tests.
Or another way would be to remove the include of `Arduino.h`.

**Review by pkendall64 — APPROVED** [review]
Looks good 👍

File: src/lib/LUA/tx_devLUA.cpp [hunk 1]
```diff
@@ -20,9 +20,9 @@
     "D50Hz(-112dBm);25Hz(-123dBm);50Hz(-120dBm);100Hz(-117dBm);100Hz Full(-112dBm);200Hz(-112dBm)"
 #elif defined(RADIO_LR1121)
 #define STR_LUA_PACKETRATES \
-    "50Hz 2.4G;100Hz Full 2.4G;150Hz 2.4G;250Hz 2.4G;333Hz Full 2.4G;500Hz 2.4G;" \
-    "50Hz Low Band;100Hz Low Band;100Hz Full Low Band;200Hz Low Band;200Hz Full Low Band;250Hz Low Band;" \
-    "X100Hz Full(-112dBm);X150Hz(-112dBm)"
+    "K1000 Full Low Band;DK500 2.4G;200 Full Low Band;250 Low Band;X100 Full;X150;" \
+    "50 2.4G;100 Full 2.4G;150 2.4G;250 2.4G;333 Full 2.4G;500 2.4G;" \
+    "50 Low Band;100 Low Band;100 Full Low Band;200 Low Band"
```
Comment (CapnBry) [inline src/lib/LUA/tx_devLUA.cpp:25]: lmao at this every time

**Review by CapnBry — APPROVED** [review]
Fudging 💪 work, Jye. Nice.

**Pull request merged**